### ISSUE TITLE
weighted_mean with group_by does not work

### ISSUE THREAD

**Issue by ftobin** [pwwang/datar#133]
numpy's average function does not handle grouped series.  I feel like an older version of datar handled this problem fine, but I can't figure out what it was doing (looking at old versions).

Example:
```
(datar.datasets.iris
  >> datar.all.group_by(f.Species)
  >> datar.all.summarize(mymean = datar.all.weighted_mean(f.Sepal_Length, f.Sepal_Width))

Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
```

In terms of fixes, `weighted_mean()` gets two grouped series, and I'm not sure the best approach to aggregate their data together.
Ideally one could just say:
```
(x + w).sum()/w.sum()
```
But you can't do arithmetic on grouped series.

**Comment by pwwang** [comment]
Did you mean to ungroup `x` and `w` first?

**Comment by pwwang** [comment]
> Things like weighted means are so clumsy in regular pandas!

Agree. That's why this package exists.

Fixed at [v0.8.6](https://github.com/pwwang/datar/releases/tag/0.8.6)
